server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -325,9 +322,6 @@ Index[] concreteIndices(Context context, String... indexExpressions) {
         final Set<Index> concreteIndicesResult = Sets.newLinkedHashSetWithExpectedSize(expressions.size());
         final Map<String, IndexAbstraction> indicesLookup = context.getState().metadata().getIndicesLookup();
         for (String expression : expressions) {
-            if (context.getOptions().ignoreUnavailable() == false) {
-                ensureAliasOrIndexExists(context, expression);
-            }
```

Comment:
This is now redundan, because `resolveExpressions` on line 323 above handles the same.
The removal avoid the checks over a potentially long list resulting from wildcard expansion, which are all redundant checks (because wildcards would not expand to anything that breaks this check).